twitter/twitter.php

```diff
@@ -1164,48 +1190,31 @@ function twitter_fetch_contact($uid, $data, $create_user)
 	return $contact_id;
 }
 
-function twitter_fetchuser(App $a, $uid, $screen_name = "", $user_id = "")
+/**
+ * @param string $screen_name
+ * @return stdClass|null
+ * @throws Exception
+ */
+function twitter_fetchuser($screen_name)
 {
 	$ckey = DI::config()->get('twitter', 'consumerkey');
 	$csecret = DI::config()->get('twitter', 'consumersecret');
-	$otoken = DI::pConfig()->get($uid, 'twitter', 'oauthtoken');
-	$osecret = DI::pConfig()->get($uid, 'twitter', 'oauthsecret');
-
-	$r = q("SELECT * FROM `contact` WHERE `self` = 1 AND `uid` = %d LIMIT 1",
-		intval($uid));
-
-	if (DBA::isResult($r)) {
-		$self = $r[0];
-	} else {
-		return;
-	}
 
-	$parameters = [];
-
-	if ($screen_name != "") {
-		$parameters["screen_name"] = $screen_name;
-	}
-
-	if ($user_id != "") {
-		$parameters["user_id"] = $user_id;
-	}
-
-	// Fetching user data
-	$connection = new TwitterOAuth($ckey, $csecret, $otoken, $osecret);
 	try {
+		// Fetching user data
+		$connection = new TwitterOAuth($ckey, $csecret);
```

Comment:
We don't need `$otoken` and `$osecret` here?

Reply:
Nope, because we don't retrieve data specific to a user.

Reply:
When its working then I'm fine with that.